src/gui/sdlmain.cpp

```diff
@@ -3307,16 +3196,44 @@ static SDL_Rect calc_viewport_fit(const int win_w, const int win_h)
 	// limit the window to the user's desired viewport, if configured
 	const auto [lwin_w, lwin_h] = restrict_to_viewport_resolution(win_w, win_h);
 
-	// calculate the aspect ratios of the draw buffer and the limited window
+	// calculate the aspect ratio of the draw buffer
 	const auto draw_aspect = (sdl.draw.width * sdl.draw.scalex) /
 	                         (sdl.draw.height * sdl.draw.scaley);
-	const auto lwin_aspect = static_cast<double>(lwin_w) / lwin_h;
 
-	// calculate the viewport contingent on the aspect ratio of the limited
-	// window versus draw buffer
-	const auto lwin_is_wider = lwin_aspect > draw_aspect;
-	const auto view_w = lwin_is_wider ? iround(lwin_h * draw_aspect) : lwin_w;
-	const auto view_h = lwin_is_wider ? lwin_h : iround(lwin_w / draw_aspect);
+	int view_w = 0;
+	int view_h = 0;
+	switch (sdl.integer_scale_mode) {
+	case INTEGER_SCALE_MODE::OFF: {
+		// calculate the viewport contingent on the aspect ratio of the
+		// limited window versus draw buffer
+		const auto lwin_aspect   = static_cast<double>(lwin_w) / lwin_h;
+		const auto lwin_is_wider = lwin_aspect > draw_aspect;
```

Comment:
From this I can guess that `draw_aspect` must be aspect ratio corrected... right?